### PR TITLE
Slider - new features

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
Slider - Adding 'reset' function with callback 'onReset', initial values to range slider, 'useGap' prop.

## Changelog
Slider - Adding 'reset' function with callback 'onReset', initial values to range slider, 'useGap' prop.

### CODE REVIEW

File: demo/src/screens/componentScreens/SliderScreen.tsx [hunk 1]
```diff
@@ -124,7 +135,28 @@ export default class SliderScreen extends Component<SliderScreenProps, SliderScr
...
           <Text $textDefault text70BO marginV-15>
-            Range slider
+            Range Slider
+          </Text>
+          <View row spread style={this.getReverseStyle()}>
+            <Text bodySmall $textNeutral>
+              min. {sliderMinValue}%
+            </Text>
+            <Text bodySmall $textNeutral>
+              max. {sliderMaxValue}%
```
Comment (lidord-wix) [inline demo/src/screens/componentScreens/SliderScreen.tsx:145]: The presented values of the two range sliders are the same (even though the values are different)..

**Review by lidord-wix — COMMENTED** [review]
Looks good!
I wrote some small comments

File: src/components/slider/index.tsx [hunk 2]
```diff
@@ -371,7 +418,8 @@ export default class Slider extends PureComponent<SliderProps, State> {
       const left = trackSize.width === 0 ? _x : (_x * nonOverlappingTrackWidth) / trackSize.width; // do not render above prefix\suffix icon\text
       
       const maxThumbPosition = this._thumbStyles?.left as number;
-      if (left < maxThumbPosition - thumbSize.width - MIN_RANGE_GAP) {
+      if (useGap && left < maxThumbPosition - thumbSize.width - MIN_RANGE_GAP
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/slider/index.tsx:421]: The gap is not actually 4 in value as it is calculated related to the thumb's width, the 4 is just adding it visually. I thought about letting the user pass the gap and decided it is unnecessary, whether you want the thumbs to overlap or not.

File: demo/src/screens/componentScreens/SliderScreen.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply demo/src/screens/componentScreens/SliderScreen.tsx:145]: I'll take a look. Thanks

Comment (Inbal-Tish) [reply demo/src/screens/componentScreens/SliderScreen.tsx]: Done

File: demo/src/screens/componentScreens/SliderScreen.tsx [hunk 3]
```diff
@@ -56,170 +65,229 @@ export default class SliderScreen extends Component<SliderScreenProps, SliderScr
...
           </Text>
-          <View row spread style={this.getReverseStyle()}>
-            <Text bodySmall $textNeutral>
-              min. {sliderMinValue}%
-            </Text>
-            <Text bodySmall $textNeutral>
-              max. {sliderMaxValue}%
-            </Text>
-          </View>
+          <Button link label="Reset" onPress={this.resetSlider}/>
```
Comment (lidord-wix) [inline demo/src/screens/componentScreens/SliderScreen.tsx:78]: The `Reset` button resets the presented value but not the slider's value

Comment (Inbal-Tish) [reply demo/src/screens/componentScreens/SliderScreen.tsx]: Fixed

File: demo/src/screens/componentScreens/SliderScreen.tsx [hunk 4]
```diff
@@ -50,141 +80,231 @@ export default class SliderScreen extends Component<SliderScreenProps, SliderScr
...
-            showLabels
-            // onValueChange={this.onGroupValueChange}
-          />
+          {/* {this.renderDefaultSliderExample()}
+          {this.renderNegativeSliderExample()}
+          {this.renderDisabledSliderExample()} */}
+          {this.renderRangeSliderExample()}
+          {/* {this.renderRangeSliderWithValuesExample()}
+          {this.renderGradientSlidersExample()}
+          {this.renderColorSliderGroupExample()} */}
```
Comment (lidord-wix) [inline demo/src/screens/componentScreens/SliderScreen.tsx:307]:
```suggestion
          {this.renderDefaultSliderExample()}
          {this.renderNegativeSliderExample()}
          {this.renderDisabledSliderExample()}
          {this.renderRangeSliderExample()}
          {this.renderRangeSliderWithValuesExample()}
          {this.renderGradientSlidersExample()}
          {this.renderColorSliderGroupExample()}
```

**Review by lidord-wix — APPROVED** [review]
Approved, please see my last comment before merging

File: demo/src/screens/componentScreens/SliderScreen.tsx (hunk 4 above)
Comment (Inbal-Tish) [reply demo/src/screens/componentScreens/SliderScreen.tsx]: Thanks!

**Comment by ajp8164** [comment]
Thanks for this! Just getting to look at these new changes - will check against my implementation needs. Thanks for linking over my PRs.

**Pull request merged**